### PR TITLE
Use DB table as cache for update monitor

### PR DESCRIPTION
**Pull request by adamiak**
Resolves L2B-2748

Uses the new reorgSafeDepth for reorg protection.

### CODE REVIEW

**Review by mateuszradomski — APPROVED** [review]
LGTM, just update the discovery version and rebase on master

File: packages/backend/package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
     "@koa/router": "^12.0.0",
     "@l2beat/backend-tools": "^0.4.0",
     "@l2beat/config": "*",
-    "@l2beat/discovery": "0.16.2",
+    "@l2beat/discovery": "0.17.0",
```
[the remark this answers is not in the document]
Comment (sdlyy) [reply packages/backend/package.json]: `19.0.0` just released :trollface:

File: packages/backend/src/core/discovery/DiscoveryCache.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  async get(key: string): Promise<string | undefined> {
+    const record = await this.repository.findByKey(key)
+    return record?.value
+  }
+
+  async set(
+    key: string,
+    value: string,
+    chainId: number,
+    blockNumber: number | undefined,
```
Comment (sdlyy) [inline packages/backend/src/core/discovery/DiscoveryCache.ts:20]: Circling back to our today's little change - let's make it required

File: packages/backend/src/core/discovery/DiscoveryCache.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  ): Promise<void> {
+    await this.repository.addOrUpdate({
+      key,
+      value,
+      chainId: ChainId(chainId),
+      blockNumber: blockNumber ?? null,
```
Comment (sdlyy) [inline packages/backend/src/core/discovery/DiscoveryCache.ts:26]: The same applies here - block number is now required

File: packages/backend/src/peripherals/database/DiscoveryCacheRepository.test.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      blockNumber: 2_000_000,
+    })
+
+    await repository.addOrUpdate(record1)
+    await repository.addOrUpdate(record2)
+    const actual = await repository.getAll()
+    expect(actual).toEqual([record2])
+  })
+
+  it('preserves nullish block number', async () => {
```
Comment (sdlyy) [inline packages/backend/src/peripherals/database/DiscoveryCacheRepository.test.ts:43]: Not needed any more I suppose 😄

**Review by sdlyy — APPROVED** [review]
GJ, please revise and correct all other places where block number is still nullable 🚀

**Pull request merged**